cognite/neat/examples/workflows/fast_graph/workflow.py

```diff
@@ -0,0 +1,311 @@
+import contextlib
+import logging
+import time
+from pathlib import Path
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import AssetFilter
+from prometheus_client import Gauge
+
+from cognite.neat.core import loader, parser
+from cognite.neat.core.data_classes.transformation_rules import TransformationRules
+from cognite.neat.core.extractors.labels import upload_labels
+from cognite.neat.core.extractors.rdf_to_assets import categorize_assets, rdf2assets, upload_assets
+from cognite.neat.core.extractors.rdf_to_relationships import (
+    categorize_relationships,
+    rdf2relationships,
+    upload_relationships,
+)
+from cognite.neat.core.loader.graph_store import NeatGraphStore, drop_graph_store
+from cognite.neat.core.transformer import RuleProcessingReport, domain2app_knowledge_graph
+from cognite.neat.core.validator import validate_asset_hierarchy
+from cognite.neat.core.workflow import utils
+from cognite.neat.core.workflow.base import BaseWorkflow, FlowMessage
+from cognite.neat.core.workflow.cdf_store import CdfStore
+
+with contextlib.suppress(ValueError):
+    prom_cdf_resource_stats = Gauge(
+        "neat_fast_graph_wf_cdf_resource_stats",
+        "CDF resource stats before and after running fast_graph workflow",
+        ["resource_type", "state"],
+    )
+with contextlib.suppress(ValueError):
+    prom_data_issues_stats = Gauge("neat_fast_graph_wf_data_issues", "Data validation issues", ["type"])
+
+
+class FastGraphNeatWorkflow(BaseWorkflow):
+    def __init__(self, name: str, client: CogniteClient):
+        super().__init__(name, client, [])
+        self.dataset_id: int = 0
+        self.current_step: str = None
+        self.source_graph: NeatGraphStore = None
+        self.solution_graph: NeatGraphStore = None
+        self.transformation_rules: TransformationRules = None
+        self.stop_on_error = False
+        self.count_create_assets = 0
+        self.graph_source_type = "memory"
+
+    def step_load_transformation_rules(self, flow_msg: FlowMessage = None):
```

Comment:
Not for this PR, but is it an idea to create a small library of steps that can be reused? This looks like a duplicated code from one of the other workflows.

Reply:
Yes, there is an idea of creating library and not just a library but make these standard steps available via workflow manifest/UI in addition to modules that can be used from code . It's evolution.